### PR TITLE
feat(HACBS-1638): upload test coverage report

### PR DESCRIPTION
**Pull request by jinqi7**

### CODE REVIEW

File: .github/workflows/pr-checks.yaml [hunk 1]
```diff
@@ -150,24 +150,12 @@ jobs:
 
       - name: Run unittests for policies
         run: |
-          /usr/bin/opa test policies unittests unittests/test_data -c > /tmp/policy_unittest_result.json
+         TEST_FILES="./test ./policies ./unittests ./unittests/test_data"
```
Comment (MartinBasti) [inline .github/workflows/pr-checks.yaml:153]: `./unittests/test_data` data doesn't contain any tests, they should be probably omitted here

**Comment by MartinBasti** [comment]
https://app.codecov.io/gh/redhat-appstudio/hacbs-test/commit/1c8228cf0871c9c12aad5f7b9f1b13727f185e0c  it doesn't work, it say invalid coverage data

I also see this in coverage report
```
<<<<<< network
# path=coverage.json
4 errors occurred:
hack/simplecov.rego:8: rego_unsafe_var_error: var obj is unsafe
hack/simplecov.rego:8: rego_unsafe_var_error: var file is unsafe
hack/simplecov.rego:13: rego_unsafe_var_error: var cm is unsafe
hack/simplecov.rego:21: rego_unsafe_var_error: var ncm is unsafe
<<<<<< EOF
```

**Review by jsztuka — APPROVED** [review]
Sorry didn't saw the previous comment.

File: .github/workflows/pr-checks.yaml (hunk 1 above)
Comment (hongweiliu17) [reply .github/workflows/pr-checks.yaml:153]: I think `./unittests/test_data` is the tested data, so we need load it in `opa test` command.

Comment (MartinBasti) [reply .github/workflows/pr-checks.yaml:153]: ah sorry, you are right, it's not related to coverage itself

**Comment by jinqi7** [comment]
In fact, the file hack/simplecov.rego is not our home made. Let me check it . Thanks


> https://app.codecov.io/gh/redhat-appstudio/hacbs-test/commit/1c8228cf0871c9c12aad5f7b9f1b13727f185e0c it doesn't work, it say invalid coverage data
> 
> I also see this in coverage report
> 
> ```
> <<<<<< network
> # path=coverage.json
> 4 errors occurred:
> hack/simplecov.rego:8: rego_unsafe_var_error: var obj is unsafe
> hack/simplecov.rego:8: rego_unsafe_var_error: var file is unsafe
> hack/simplecov.rego:13: rego_unsafe_var_error: var cm is unsafe
> hack/simplecov.rego:21: rego_unsafe_var_error: var ncm is unsafe
> <<<<<< EOF
> ```

**Review by MartinBasti — CHANGES_REQUESTED** [review]
It shows coverage now :)

This runs coverage only for PRs, but we need also base coverage for main branch, otherwise coverage decrease won't be reported by codecov

**Comment by jinqi7** [comment]
> This runs coverage only for PRs, but we need also base coverage for main branch, otherwise coverage decrease won't be reported by codecov
The coverage is restricted to 100% from the script in this repo.  And from the experience of adding codecov for integration-service, the base coverage should be there after being uploaded this time. @srivickynesh ,is it right?

**Comment by MartinBasti** [comment]
> > This runs coverage only for PRs, but we need also base coverage for main branch, otherwise coverage decrease won't be reported by codecov
> > The coverage is restricted to 100% from the script in this repo.  And from the experience of adding codecov in PR [redhat-appstudio/integration-service#31](https://github.com/redhat-appstudio/integration-service/pull/31) for integration-service, the base coverage should be there after being uploaded this time. @srivickynesh , is it right?

in PR you mentioned, there is an extra workflow https://github.com/redhat-appstudio/integration-service/pull/31/files#diff-512f30627f95d7fcdc76dc362d896351e9a4539994b7ab00b41979e653a6d0a9  which runs for pushes (mereges) to main, to generate base coverage from main branch. Such workflow is missing in this PR.

Also I'd suggest to avoid duplicating workflows and just run the same workflow also for push actions

**Comment by jinqi7** [comment]
@MartinBasti Can you please help to review again?

**Review by MartinBasti — APPROVED** [review]

**Pull request merged**